artmaker/src/main/java/io/artmaker/ArtMakerViewModel.kt

```diff
@@ -83,45 +87,30 @@ internal class ArtMakerViewModel(
     private val _finishedImage = MutableStateFlow<Bitmap?>(null)
     val finishedImage = _finishedImage.asStateFlow()
 
+    init {
+        listenToUndoRedoState()
+    }
+
     fun onAction(action: ArtMakerAction) {
         when (action) {
             is ArtMakerAction.TriggerArtExport -> triggerArtExport(action.type)
             is ArtMakerAction.ExportArt -> exportArt(action.bitmap)
-            ArtMakerAction.Redo -> redo()
-            ArtMakerAction.Undo -> undo()
-            ArtMakerAction.Clear -> clear()
             ArtMakerAction.UpdateBackground -> updateBackgroundColour()
             is ArtMakerAction.SelectStrokeColour -> updateStrokeColor(colour = action.color)
-            is ArtMakerAction.SelectStrokeWidth -> selectStrokeWidth(strokeWidth = action.strokeWidth)
+            is ArtMakerAction.SetStrokeWidth -> selectStrokeWidth(strokeWidth = action.strokeWidth)
+            is ArtMakerAction.UpdateSetStylusOnly -> updateStylusSetting(useStylusOnly = action.shouldUseStylusOnly)
+            is ArtMakerAction.UpdateStylusDialogNeverShow -> updateStylusDialogShowSetting(canShowStylusDialog = action.canShowStylusDialog)
         }
     }
 
-    fun onDrawEvent(event: DrawEvent) {
-        when (event) {
-            is DrawEvent.AddNewShape -> addNewShape(event.offset)
-            DrawEvent.UndoLastShapePoint -> undoLastShapePoint()
-            is DrawEvent.UpdateCurrentShape -> updateCurrentShape(event.offset)
-        }
-    }
-
-    private fun addNewShape(offset: Offset) {
-        val data = PointsData(
-            points = mutableStateListOf(offset),
-            strokeColor = Color(artMakerUIState.value.strokeColour),
-            strokeWidth = artMakerUIState.value.strokeWidth.toFloat(),
-        )
-        _pathList.add(data)
-        _artMakerUIState.update { it.copy(canUndo = true, canClear = true) }
-    }
-
-    private fun updateCurrentShape(offset: Offset) {
-        val idx = _pathList.lastIndex
-        _pathList[idx].points.add(offset)
-    }
+    fun onDrawEvent(event: DrawEvent) = drawingManager.onDrawEvent(event, _uiState.value.strokeColour, _uiState.value.strokeWidth)
```

Comment:
The drawing manager now handles all draw events

Reply:
:+1:
:thumbsup:

Reply:
This is well noted...